model-scoring-core/src/main/scala/org/trustedanalytics/scoring/ScoringService.scala

```diff
@@ -1,97 +1,103 @@
+  */
 package org.trustedanalytics.scoring
 
 import java.io.File
-import akka.actor.{Props,Actor}
+
+import akka.actor.{Actor, Props}
 import org.apache.commons.io.FileUtils
 import org.trustedanalytics.model.archive.format.ModelArchiveFormat
 import spray.routing._
 import spray.http._
 import spray.http.BodyPart
 import MediaTypes._
 import akka.event.Logging
+import com.typesafe.config.ConfigFactory
+
 import scala.concurrent._
 import ExecutionContext.Implicits.global
-import scala.util.{ Failure, Success }
+import scala.util.{Failure, Success}
 import org.trustedanalytics.scoring.interfaces.Model
 import spray.json._
+import DefaultJsonProtocol._
+import org.trustedanalytics.scoring.ScoringServiceJsonProtocol._
+import spray.routing.directives.RouteDirectives
 
 
 /**
- * We don't implement our route structure directly in the service actor because
- * we want to be able to test it independently, without having to spin up an actor
- *
- * @param scoringService the service to delegate to
- */
+  * We don't implement our route structure directly in the service actor because
+  * we want to be able to test it independently, without having to spin up an actor
+  *
+  * @param scoringService the service to delegate to
+  */
 class ScoringServiceActor(val scoringService: ScoringService) extends Actor with HttpService {
 
   /**
-   * the HttpService trait defines only one abstract member, which
-   * connects the services environment to the enclosing actor or test
-   */
+    * the HttpService trait defines only one abstract member, which
+    * connects the services environment to the enclosing actor or test
+    */
   override def actorRefFactory = context
 
   /**
-   * Delegates to Scoring Service.
-   *
-   * This actor only runs our route, but you could add other things here, like
-   * request stream processing or timeout handling
-   */
+    * Delegates to Scoring Service.
+    *
+    * This actor only runs our route, but you could add other things here, like
+    * request stream processing or timeout handling
+    */
   def receive = runRoute(scoringService.serviceRoute)
 }
 
 /**
- * Defines our service behavior independently from the service actor
- */
+  * Defines our service behavior independently from the service actor
+  */
 class ScoringService(model: Model) extends Directives {
-  var scoringModel = model
 
-  def homepage = {
-    respondWithMediaType(`text/html`) {
-      complete {
-        <html>
-          <body>
-            <h1>Welcome to the New Scoring Engine</h1>
-          </body>
-        </html>
-      }
-    }
-  }
+  import spray.json._
+
+  var scoringModel = model
 
   lazy val description = {
     new ServiceDescription(name = "Trusted Analytics",
       identifier = "ia",
       versions = List("v1", "v2"))
   }
 
-  val jsonFormat = new ScoringServiceJsonProtocol()
-  import jsonFormat._
+  /**
+    * Case class to hold model, model tar file path and json format protocol object
```

Comment:
Yes, that's correct. The latest scoring engine do not take path from HDFS, hence we no longer hold the tar file path.